core/src/com/fallenflame/game/LevelController.java

```diff
@@ -1,5 +1,498 @@
+            enemy.dispose();
+            enemies.clear();
+        }
+        for(FlareModel flare : flares) {
+            flare.deactivatePhysics(world);
+            flare.dispose();
+            flares.clear();
+        }
+        exit.deactivatePhysics(world);
+        exit.dispose();
+        player.deactivatePhysics(world);
+        player.dispose();
+
+        if (world != null) {
+            world.dispose();
+            world = null;
+        }
+    }
+
+    /**
+     * Returns true if the object is in bounds.
+     *
+     * This assertion is useful for debugging the physics.
+     *
+     * @param obj The object to check.
+     *
+     * @return true if the object is in bounds.
+     */
+    private boolean inBounds(Obstacle obj) {
+        boolean horiz = (bounds.x <= obj.getX() && obj.getX() <= bounds.x+bounds.width);
+        boolean vert  = (bounds.y <= obj.getY() && obj.getY() <= bounds.y+bounds.height);
+        return horiz && vert;
+    }
+
+    /**
+     * Updates all of the models in the level.
+     *
+     * This is borderline controller functionality.  However, we have to do this because
+     * of how tightly coupled everything is.
+     *
+     * @param dt the time passed since the last frame
+     */
+    public void update(float dt) {
+        // Update player (and update levelModel) and exit
+        levelModel.removePlayer(player);
+        player.update(dt);
+        levelModel.placePlayer(player);
+        exit.update(dt); // TODO: Necessary?
+
+        // Get Enemy Actions
+        Iterator<AIController> ctrlI = AIControllers.iterator();
+        LinkedList<EnemyModel.Action> actions = new LinkedList();
+        while(ctrlI.hasNext()){
+            AIController ctrl = ctrlI.next();
+            actions.add(ctrl.getAction());
+        }
+        // Execute Enemy Actions (and update levelModel)
+        Iterator<EnemyModel> enemyI = enemies.iterator();
+        Iterator<EnemyModel> actionI = enemies.iterator();
+        while(enemyI.hasNext()){
+            EnemyModel enemy = enemyI.next();
+            levelModel.removeEnemy(enemy);
+            enemy.executeAction(actionI.next()); // TODO handle here or in enemyModel?
+            levelModel.placeEnemy(enemy);
+        }
+
+        // Update flares
+        Iterator<FlareModel> i = flares.iterator();
+        while(i.hasNext()){
+            FlareModel flare = i.next();
+            if(!flare.isActive()){
+                flare.deactivatePhysics(world);
+                flare.dispose();
+                i.remove();
+            }
+            else {
+                flare.update();
+            }
+        }
+
+        // Update lights
+        lightController.updateLights(player, flares, enemies);
+    }
+
+    /**
+     * Launch a flare from the player towards the mouse position based on preset flareJSON data
+     *
+     * @param mousePosition Position of mouse when flare launched
+     */
+    public void createFlare(float[] mousePosition){
```

Comment:
Why are you using a float array as opposed to a Vector2?

Reply:
true, I can switch it